mms/tests/unit_tests/test_client_sdk_generator.py

```diff
@@ -0,0 +1,66 @@
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+import pytest
+from mock import patch, mock_open
+from collections import namedtuple
+
+import mms.client_sdk_generator
+from mms.client_sdk_generator import ClientSDKGenerator
+
+sample_openapi_endpoints = dict()
+sample_sdk_language = 'python'
+
+@pytest.fixture()
+def patches(mocker):
+    Patches = namedtuple('Patches', ['path_exists', 'dirname', 'abspath', 'makedirs', 'json_dump', 'subprocess_call', 'open', 'logger_info'])
+    patches = Patches(
+        mocker.patch('os.path.exists'),
+        mocker.patch('os.path.dirname'),
+        mocker.patch('os.path.abspath'),
+        mocker.patch('os.makedirs'),
+        mocker.patch('json.dump'),
+        mocker.patch('subprocess.call'),
+        mocker.patch('builtins.open', mock_open()),
+        mocker.patch('mms.client_sdk_generator.logger.info')
+    )
+    patches.path_exists.return_value = True
+    patches.dirname.return_value = "testdirname"
+    patches.abspath.return_value = ""
+    patches.makedirs.return_value = ""
+    patches.json_dump.return_value = ""
+    patches.subprocess_call.return_value = ""
+    patches.logger_info.return_value = ""
+    return patches
+
+def test_handles_exception(patches):
+    test_exception = Exception("test")
+    patches.path_exists.side_effect = test_exception
+    with pytest.raises(Exception) as excinfo:
+        ClientSDKGenerator.generate(sample_openapi_endpoints, sample_sdk_language)
+    patches.json_dump.assert_not_called()
+    patches.subprocess_call.assert_not_called()
+    assert excinfo.value != test_exception
```

Comment:
Should we actually be comparing against a dummy object ? Should we rather not compare the error message we are getting back from the exception ?

Reply:
I don't want to copy the contents of the error message text in the test.  What really should happen (and we do elsewhere), is that we use the MSError class.  Then we can check whether the exception returned is an MSError and potentially that it uses the correct error code.  Here, I just ensure that the exception was handled in some fashion because it raises a new exception

Reply:
I don't want to copy the contents of the error message text in the test. What really should happen (and we do elsewhere), is that we use the MSError class. Then we can check whether the exception returned is an MSError and potentially that it uses the correct error code. Here, I just ensure that the exception was handled in some fashion because it raises a new exception